app/models/shop.rb

```diff
@@ -1,2 +1,4 @@
 class Shop < ActiveRecord::Base
+  has_many :shop_items
+  has_many :items, through: :shop_items
```

Comment:
In here is where we would set up the slug. Assuming we have the slug column, we could write something like this:

```ruby
  before_validation :generate_slug

  private

  def generate_slug
    self.slug = name.parameterize
  end
```